File: src/main/java/ua/softgroup/matrix/server/service/TrackingDataService.java
package ua.softgroup.matrix.server.service;

import ua.softgroup.matrix.api.model.datamodels.ActiveWindowModel;
import ua.softgroup.matrix.server.persistent.entity.TrackingData;

import java.util.List;

public interface TrackingDataService extends GeneralEntityService<TrackingData> {

    void saveTrackingData(String userToken, Long projectId, String keyboardText, Double mouseFootage,
                          List<ActiveWindowModel> activeWindowList, byte[] screenshot, String screenshotTitle);

}

File: src/main/scala/ua/softgroup/matrix/server/service/TrackingDataService.scala
package ua.softgroup.matrix.server.service

import ua.softgroup.matrix.api.model.datamodels.ActiveWindowModel
import ua.softgroup.matrix.server.persistent.entity.TrackingData

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait TrackingDataService extends GenericEntityService[TrackingData] {

  def saveTrackingData(userToken: String, projectId: Long, keyboardText: String, mouseFootage: Double,
                       activeWindowList: List[ActiveWindowModel], screenshot: Array[Byte], screenshotTitle: String)

}
